### ISSUE TITLE
Random mouse freezes

### ISSUE THREAD

**Issue by gauravsaini964** [xiaoMGitHub/LEGION_Y7000Series_Hackintosh#76]
I have legion y530 i7, gtx 1050. My trackpad freezes occasionally and it comes back to life after a minute or so automatically.
Same issue persists on windows but others have solved it by installing synaptic drivers.
Is I2C GPIO related to it ?

What can be done for macOS ?

@xiaoMGitHub Btw, thanks for releasing 3.0.2.
PS: Please consider moving development group to telegram or much more global platform, we would love to donate and help in development as well.

**Comment by RaviKavaiya** [comment]
@gauravsaini964 I agree to your idea of moving to Telegram...

**Comment by xiaoMGitHub** [comment]
Due to network restrictions, it was very difficult for me to use telegram. For EFI, my original intention was to adapt the customized model y7000, not the international version y530 / y540.

**Comment by gauravsaini964** [comment]
Alright now issue then we will wait for delayed release...
Right now I am facing issue on fresh install...
 I have 2 SSD
NVME - Windows
Sata SSD - macOS ( OC 3.0.1 by you)

I created bootable USB having OC 3.0.2, rebooted and selected USB to boot..
Installer gives me this error:

`00:000 00:000 AAPL: [EB|#LOG:INIT] 2020-04-25T22:51:35
00:101 00:101 AAPL: [EB|#VERSION] bootbase.efi 422.100.16~12 (Official), built 2020-02-29T00:59:07-0800
00:194 00:092 AAPL: [EB|#BUILD] BUILD-INFO[310]:{"DisplayName":"bootbase.efi","DisplayVersion":"422.100.16~12","RecordUuid":"FD85960B-6B0F-41D5-8944-A1E80707DC36","BuildTime":"2020-02-29T00:59:07-0800","ProjectName":"efiboot","ProductName":"bootbase.efi","SourceVersion":"422.100.16","BuildVersion":"12","BuildConfiguration":"Release","BuildType":"Official"}
00:287 00:093 AAPL: [EB|`H:IS] Err(0xE) @ GV boot-signature
00:406 00:119 AAPL: [EB|`H:IS] Err(0xE) @ GV boot-image-key
00:478 00:071 AAPL: [EB|#H:IS] 0
00:572 00:093 AAPL: [EB|#WL:MODE] 0
00:666 00:093 AAPL: [EB|#CONFIG:SRC] <default>
00:760 00:094 AAPL: [EB|#CONFIG| 0] boot-save-log : 0x0000000000000002 (0x0000000000000002 < 0xFFFFFFFFFFFFFFFF) <default>
00:833 00:072 AAPL: [EB|#CONFIG| 1] wake-save-log : 0x0000000000000002 (0x0000000000000002 < 0x0000000000000002) <default>
00:927 00:094 AAPL: [EB|#CONFIG| 2] console       : 0x0000000000000001 (0x0000000000000001 < 0x0000000000000001) <default>
01:021 00:093 AAPL: [EB|#CONFIG| 3] serial        : 0x0000000000000000 (0x0000000000000000 < 0x0000000000000000) <default>
01:116 00:094 AAPL: [EB|#CONFIG| 4] embed-log-dt  : 0x0000000000000000 (0x0000000000000000 < 0x0000000000000000) <default>
01:210 00:094 AAPL: [EB|#CONFIG| 5] timestamps    : 0x0000000000000000 (0x0000000000000000 < 0xFFFFFFFFFFFFFFFF) <default>
01:308 00:097 AAPL: [EB|#CONFIG| 6] log-level     : 0x0000000000000001 (0x0000000000000001 & 0x0000000000000001) <default>
01:373 00:065 AAPL: [EB|#CONFIG| 7] breakpoint    : 0x0000000000000000 (0x0000000000000000 & 0x0000000000000000) <default>
01:438 00:064 AAPL: [EB|#CONFIG| 8] kc-read-size  : 0x0000000000100000 (0x0000000000100000 < 0xFFFFFFFFFFFFFFFF) <default>
01:502 00:063 AAPL: [EB|#WL|0] 0x01 0x01   0 0x00
01:597 00:094 AAPL: [EB|#BRD:NV] Mac-27ADBB7B4CEE8E61
01:688 00:091 AAPL: [EB|`B:IEP] Err(0xE) @ LocP 8ECE08D8-A6D4-430B-A7B0-2DF318E7884A
01:790 00:101 AAPL: [EB|#WL|1] 0x01 0x01   2 0x00
01:882 00:092 AAPL: [EB|`BST:IDT] {
01:976 00:093 AAPL: [EB|`BST:IDT] }
02:044 00:068 AAPL: [EB|#WL|2] 0x01 0x01   3 0x00
02:140 00:095 AAPL: [EB|#WL|3] 0x01 0x01   4 0x00
02:218 00:078 AAPL: [EB|#BRD:NV] Mac-27ADBB7B4CEE8E61
02:304 00:085 AAPL: [EB|#WL|4] 0x01 0x01   5 0x00
02:398 00:093 AAPL: [EB|`H:CHK] BM: 0x0000000000000000
02:490 00:092 AAPL: [EB|`H:CHK] Err(0xE) @ GV boot-signature
02:555 00:064 AAPL: [EB|#WL|5] 0x01 0x01  23 0x0E
02:641 00:086 AAPL: [EB|`H:CPM] Err(0xE), (24, 24) @ GV Persistent-memory-note
02:706 00:065 AAPL: [EB|`SB:EN] Err(0xE) @ LocP 24B73556-2197-4702-82A8-3E1337DAFBF2
02:771 00:064 AAPL: [EB|#LIMG:DP] Acpi(PNP0A03,0)/Pci(14|0)/Usb(2, 0)/HD(Part2,Sig5A813D0E-2BEC-485F-8065-66082AAA5FB2)
02:835 00:064 AAPL: [EB|#LIMG:FP] \System\Library\CoreServices\boot.efi
02:919 00:083 AAPL: [EB|#LIMG:OPT] ()
03:013 00:093 AAPL: [EB|#RPS] R?0 P?0 S?0
03:107 00:094 AAPL: [EB|`OPT:LXF] 'LF (\\System\\Library\\CoreServices\\com.apple.Boot.plist)
03:236 00:128 AAPL: [EB|`LD:LF] FIO: 0, DIR: 1, P: \\System\\Library\\CoreServices\\com.apple.Boot.plist, DP: 0
03:334 00:097 AAPL: [EB|`LD:OFS] Err(0xE) @ OPEN (\\System\\Library\\CoreServices\\com.apple.Boot.plist)
03:427 00:093 AAPL: [EB|`OPT:LXF] Err(0xE) @ 'LF
03:522 00:094 AAPL: [EB|`OPT:LXF] 'LF (Library\\Preferences\\SystemConfiguration\\com.apple.Boot.plist)
03:616 00:094 AAPL: [EB|`LD:LF] FIO: 0, DIR: 1, P: Library\\Preferences\\SystemConfiguration\\com.apple.Boot.plist, DP: 0
03:716 00:099 AAPL: [EB:#KF] (root-dmg=file:///Install%20macOS%20Catalina.app/Contents/SharedSupport/BaseSystem.dmg)
03:809 00:093 AAPL: [EB|#MBA:CL] ()
03:903 00:094 AAPL: [EB|#MBA:NV] (-v keepsyms=1 debug=0x100 alcid=1)
03:998 00:094 AAPL: [EB|#MBA:KF] (root-dmg=file:///Install%20macOS%20Catalina.app/Contents/SharedSupport/BaseSystem.dmg)
04:097 00:099 AAPL: [EB|#MBA:OUT] (-v keepsyms=1 debug=0x100 alcid=1 root-dmg=file:///Install%20macOS%20Catalina.app/Contents/SharedSupport/BaseSystem.dmg)
04:191 00:094 AAPL: [EB|#LOG:VERBOSE] 2020-04-25T22:51:39
04:285 00:094 AAPL: [EB|#OPT:BM] 0x4082
04:370 00:084 AAPL: [EB|`OPT:LXF] 'LF (\\System\\Library\\CoreServices\\PlatformSupport.plist)
04:448 00:078 AAPL: [EB|`LD:LF] FIO: 0, DIR: 1, P: \\System\\Library\\CoreServices\\PlatformSupport.plist, DP: 0
04:545 00:096 AAPL: [EB|#B:VAw]
04:637 00:092 AAPL: [EB|`P:DCCPR] Err(0xE) @ GV AAPL,CoProcessorReboot
04:731 00:093 AAPL: [EB|`P:DCMOSP] Err(0xE) @ GV AAPL,PanicInfo0000 (0)
04:848 00:117 AAPL: [EB|`CS:CSKSD] {
04:937 00:088 AAPL: [EB|`CS:CSKSD] Err(0xE) @ GV 8D63D4FE-BD3C-4AAD-881D-86FD974BC1DF:boot-info-payload
05:005 00:068 AAPL: [EB|`G:CS] } Ok(0)
05:100 00:095 AAPL: [EB|`LD:LF] FIO: 0, DIR: 1, P: <null string>, DP: 1
05:205 00:105 AAPL: [EB|`B:SBS] SZ: 617016
05:298 00:092 AAPL: [EB|#B:SHA] 103ea7708e088c76d181d2ec1cc2e188f42c5b53
05:363 00:064 AAPL: [EB|`WL:PWLFNV] Err(0xE) @ GV wake-failure
05:426 00:063 AAPL: [EB|`WL:DT] Err(0xE) @ WL:PWLFNV
05:498 00:071 AAPL: [EB|`RTC:CP] Err(0xE) @ LocP E121EC07-9C42-45EE-B0B6-FFF8EF03C521
05:598 00:100 AAPL: [EB|#WL:DT] 0xD6 0xC7 161 0xD3
05:693 00:095 AAPL: [EB|`LD:LKC] BPDK -> (System\Library\PrelinkedKernels\prelinkedkernel.development)
05:789 00:095 AAPL: [EB|`LD:OFS] Err(0xE) @ OPEN (System\\Library\\PrelinkedKernels\\prelinkedkernel.development)
05:879 00:090 AAPL: [EB|`LD:LKC] BPDK,!R -> (System\Library\PrelinkedKernels\prelinkedkernel)
08:951 03:071 AAPL: [EB|`LD:LKFS] } Ok(0)
09:048 00:097 AAPL: [EB|`LD:LKC] } Ok(0)
09:146 00:097 AAPL: [EB|#FWFM] 0xFF1FFF3F
09:279 00:132 AAPL: [EB|#FWFT] 0xE00FE137
09:381 00:102 AAPL: [EB|#CSR:IN] 0x00000040
09:478 00:096 AAPL: [EB|#CSR:OUT] 0x00000040
09:577 00:099 AAPL: [EB|`BST:FBS] {
09:675 00:097 AAPL: [EB|`BST:FBS] BA.ADSZ: 0
09:779 00:104 AAPL: [EB|`BST:FBS] BA.KSDSZ: 0
09:878 00:098 AAPL: [EB|`B:WFDW] Err(0xE), 0 @ LocHB 71B4903C-14EC-42C4-BDC6-CE1449930E49
09:971 00:093 AAPL: [EB|#LOG:DT] 2020-04-25T22:51:45
10:038 00:067 AAPL: [EB|#LOG:EXITBS:START] 2020-04-25T22:51:45`

Any idea how to rectify this ?

**Comment by xiaoMGitHub** [comment]
Try it with CLOVER.

**Comment by gauravsaini964** [comment]
Okay,
I tried with clover it worked.
I installed macOS 10.15.4, replaced existing OC EFI with Clover EFI in my boot drive.

But since I do not like OC, I again made a bootable USB with OC 3.0.2, this time, everything worked.

Few things I noticed:
OC theme is changed and it no longer boot to any default drive after certain time. (How to fix this behaviour ?)

Cmd key was mapped to Alt key earlier now this behaviour has changed... How to revert this ?
I was used to earlier behaviour.
